include/aikido/robot/ros/RosRobot.hpp

```diff
@@ -18,21 +21,198 @@ AIKIDO_DECLARE_POINTERS(RosRobot)
+      std::string desiredName);
+
+  ///
+  /// Loads the controller. Throws a runtime_error if controller cannot be
+  /// loaded. Add an (executor, controller) to the inactive (executor,
+  /// controller, control mode) list Operations on this executor do not affect
+  /// the control mode. Releases DoFs held by executor if held.
+  ///
+  /// \param[in] executor The Executor to add to the inactive list.
+  /// \param[in] controllerName The corresponding controller to load and add to
+  /// the inactive list. \return A robot-unique non-negative ID (negative
+  /// implies failure)
+  std::string registerExecutor(
+      aikido::control::ExecutorPtr executor, 
+      std::string controllerName,
+      std::string desiredName);
+
+  ///
+  /// Add an executor to the inactive executors list.
+  /// Operations on this executor do not affect the controller or control mode.
+  /// Releases DoFs held by executor if held.
+  ///
+  /// \param[in] executor The Executor to add to the inactive list.
+  /// \return A robot-unique non-negative ID (negative implies failure)
+  std::string registerExecutor(aikido::control::ExecutorPtr executor, std::string desiredName = "") override;
+
+  ///
+  /// Deactivates the current active executor.
+  /// Starts the corresponding controller and switches to corresponding control
+  /// mode. Sets an executor from the inactive executor list to be the active
+  /// executor. Holds and releases DoFs as needed.
+  ///
+  /// \param[in] id of executor on executor list
+  /// \return True if successful. If false, all executors are inactive.
+  bool activateExecutor(std::string id) override; 
+
+  using Robot::activateExecutor; // This un-hides Robot::activateExecutor(const aikido::control::ExecutorType type)
+
+  ///
+  /// Sets the ros controller service client which enables controller loading.
+  ///
+  /// \param[in] rosControllerServiceClient ros controller service client.
+  void setRosLoadControllerServiceClient(
+      const std::shared_ptr<::ros::ServiceClient>& rosLoadControllerServiceClient); 
+
+  ///
+  /// Sets the ros controller service client which enables controller switching.
+  ///
+  /// \param[in] rosControllerServiceClient ros controller service client.
+  void setRosSwitchControllerServiceClient(
+      const std::shared_ptr<::ros::ServiceClient>& rosSwitchControllerServiceClient);
+
+  ///
+  /// Sets the ros joint mode command client which enables controller mode switching.
+  ///
+  /// \param[in] rosJointModeCommandClient ros joint mode command client.
+  void setRosJointModeCommandClient(
+      const std::shared_ptr<aikido::control::ros::RosJointModeCommandClient>& rosJointModeCommandClient);
+
+  ///
+  /// Starts the controller with name startControllerName.
+  ///
+  /// \param[in] startControllerName name of controller to start.
+  /// \return True if successful. Otherwise false.
+  bool startController(const std::string startControllerName);
```

Comment:
start/stop/load/switch controllers should be private functions. We want external classes only handling controllers via the executor interface.